lib/tile_bag.rb

```diff
@@ -0,0 +1,40 @@
+module Scrabble
+  class TileBag
+    attr_reader :tile_bag
+
+    def initialize
+      @tile_bag = Scrabble::TileBag.create_tile_bag
+    end
+
+    def self.create_tile_bag
+      letters = ("a".."z").to_a
+      quantity= [9,2,2,4,12,2,3,2,9,1,1,4,2,6,8,2,1,6,4,6,4,2,2,1,2,1]
+      tile_bag = []
+
+      26.times do |i|
+        quantity[i].times do
+          tile_bag<< letters[i]
+        end
+      end
+      tile_bag
+    end#end of create_tile_bag method
+
+    def draw_tiles(num)
+      if num.class != Integer
+        raise ArgumentError.new("Parameter must be integer")
+      end
+
+      drawn_tiles = []
+      num.times do
+        rand_num = rand(0...@tile_bag.length)
+        drawn_tiles << @tile_bag[rand_num]
+        @tile_bag.delete_at(rand_num)
```

Comment:
Good solution to this problem. A slightly cleaner way might be to use `Array#shuffle` and `Array#pop` - go look them up!